### PR TITLE
refactor: add response data generic type

### PR DESCRIPTION
**Pull request by Miodec**
Adds a generic to the endpoint type, allowing the api data to be properly typed

### CODE REVIEW

File: frontend/src/ts/ape/adapters/axios-adapter.ts [hunk 1]
```diff
@@ -87,6 +87,7 @@ function apeifyClientMethod(
     return {
       status: 500,
       message: errorMessage,
+      data: null,
```
Comment (Bruception) [inline frontend/src/ts/ape/adapters/axios-adapter.ts:90]: Is the intention to serializar this always? I think we should just omit.

Comment (Bruception) [reply frontend/src/ts/ape/adapters/axios-adapter.ts:90]: Nvm this is frontend code

Comment (Miodec) [reply frontend/src/ts/ape/adapters/axios-adapter.ts:90]: Just consistency with the backend - that also always returns null.

File: frontend/src/ts/ape/endpoints/users.ts [hunk 2]
```diff
@@ -246,17 +241,17 @@ export default class Users {
...
 
-  async forgotPasswordEmail(email: string): Ape.EndpointData {
+  async forgotPasswordEmail(email: string): Ape.Endpoint {
     return await this.httpClient.post(`${BASE_PATH}/forgotPasswordEmail`, {
       payload: { email },
     });
   }
 
-  async setStreakHourOffset(hourOffset: number): Ape.EndpointData {
+  async setStreakHourOffset(hourOffset: number): Ape.Endpoint {
```
Comment (Bruception) [inline frontend/src/ts/ape/endpoints/users.ts:254]: I think EndpointData makes more sense

File: frontend/src/ts/ape/types/ape.d.ts [hunk 3]
```diff
@@ -5,13 +5,15 @@ declare namespace Ape {
     payload?: any;
   }
 
-  interface HttpClientResponse {
+  interface HttpClientResponse<DataType> {
```
Comment (Bruception) [inline frontend/src/ts/ape/types/ape.d.ts:8]: Nice

File: frontend/src/ts/ape/endpoints/users.ts (hunk 2 above)
Comment (Miodec) [reply frontend/src/ts/ape/endpoints/users.ts:254]: The reason for the change is that `EndpointData` suggests a variable of this type *only* holds the data returned from the API - when in reality, it holds status, message and data:
```ts
interface Endpoint = Promise<{
    status: number;
    message: string;
    data?: any;
  }>
```

So, if you think about it its:
```ts
const endpointData: EndpointData = (api fetch)

//now we use the data somewhere
console.log(ednpointData.data); //saying data twice here
```

`EndpointResponse` would also work if you dont like just `Endpoint`

File: frontend/src/ts/elements/psa.ts [hunk 4]
```diff
@@ -20,6 +20,9 @@ function setMemory(id: string): void {
 
 async function getLatest(): Promise<MonkeyTypes.PSA[] | null> {
   const response = await Ape.psas.get();
+
+  console.log(response);
```
Comment (Bruception) [inline frontend/src/ts/elements/psa.ts:24]: Trace log

File: frontend/src/ts/ape/types/ape.d.ts [hunk 5]
```diff
@@ -5,13 +5,15 @@ declare namespace Ape {
...
     status: number;
     message: string;
-    data?: any;
+    data: DataType | null;
   }
 
-  type EndpointData = Promise<HttpClientResponse>;
+  type Endpoint<ResponseDataType = any> = Promise<
```
[the remark this answers is not in the document]
Comment (Miodec) [reply frontend/src/ts/ape/types/ape.d.ts:14]: Yeap, just suggested that in the other comment.

File: frontend/src/ts/pages/login.ts [hunk 6]
```diff
@@ -66,6 +66,8 @@ const checkNameDebounced = debounce(1000, async () => {
   }
   const response = await Ape.users.getNameAvailability(val);
 
+  console.log(response);
```
Comment (Bruception) [inline frontend/src/ts/pages/login.ts:69]: Trace log

**Review by Bruception — COMMENTED** [review]
Nice, just a comment on naming

**Review by Bruception — APPROVED** [review]

**Pull request merged**